### DISCUSSION TITLE
PiFire in Home Assistant

### DISCUSSION THREAD

**Discussion by stevenrhawk** [nebhead/PiFire#54] [General]
Good morning everyone. I just stumbled upon this awesome project, and am considering building this for my new Austin XL. Has anyone tried/succeeded in pulling PiFire into Home Assistant? I'm on a crusade to get every "smart" device in my home under a single pane of glass, and this would be a phenomenal addition. I look forward to hearing about any attempts made!

**Comment by nebhead** [comment]
> Good morning everyone. I just stumbled upon this awesome project, and am considering building this for my new Austin XL. Has anyone tried/succeeded in pulling PiFire into Home Assistant? I'm on a crusade to get every "smart" device in my home under a single pane of glass, and this would be a phenomenal addition. I look forward to hearing about any attempts made!

Hey @snackpack85 - I'm totally with you on this!  I am also a Home Assistant user and would love to get something going for this project.  Unfortunately, outside of doing the standard MQTT or API integrations (which are very basic, buttons, etc. ) I don't have too much experience creating one from scratch.  I'd like to take a crack at this some day though, or if anyone else in this community has an expertise in HA integrations, maybe we could get some help.

**Comment by nebhead** [comment]
FYI: I've tweaked the PiFire API with the latest commit to the development branch (https://github.com/nebhead/PiFire/commit/b9b91a9af4c11964f3ccf590543a7e2f4b425763).  The idea being that this can be used as a REST interface for Home Assistant.  The /api/current call will provide current temps, setpoint temps, mode, etc. so we can use this information for a template to provide gauges, etc.  And the for sending commands, we have POST /api/control to allow for rest_commands to be sent, or perhaps a REST switch.  Anyway more research and testing required.  

Found this article on the community forum that could be used to sketch out a basic interface, in a similar way: https://community.home-assistant.io/t/monitoring-a-bbq-with-ha-and-cloudbbq/217043

**Comment by nebhead** [comment]
>Is it possible to install "lite" version of PiFire on Zero, just for REST API. So I am trying to avoid getting into dependency fighting of code and its modules. So essentially pitboss-bt to rest/mqtt/x gateway.

Should definitely be possible.  Which dependencies are you concerned with?
